test/unit/math/CMakeLists.txt

```diff
@@ -23,11 +23,11 @@ target_include_directories(
 target_link_libraries(
     ${_TARGET_NAME}
     PRIVATE common)
-set_target_properties(
+target_compile_options(
     ${_TARGET_NAME}
-    PROPERTIES
-    COMPILE_OPTIONS
-    $<$<CXX_COMPILER_ID:clang>:"-ffp-contract=off"> # ffp-contract: https://llvm.org/docs/CompileCudaWithLLVM.html#id5
+    PRIVATE
+    # ffp-contract: https://llvm.org/docs/CompileCudaWithLLVM.html#flags-that-control-numerical-code
+    "$<$<COMPILE_LANG_AND_ID:CUDA,AppleClang,Clang,IntelLLVM>:SHELL:-ffp-contract=off>"
     )
```

Comment:
So this is not a problem with icpx compiling ordinary C++?

Reply:
I don't know. The previous version only seemed to care about CUDA so I did as well.